eventstore/test_tasks.py

```diff
@@ -25,8 +31,174 @@ def test_send_successful(self):
+            contact_id,
+            {
+                "helpdesk_timeout": None,
+                "wait_for_helpdesk": None,
+                "helpdesk_message_id": None,
+            },
+        )
+
+        responses.add(
+            responses.POST, f"http://turn/v1/chats/27820001001/archive", json={}
+        )
+
+        tasks.handle_expired_helpdesk_contacts()
+
+        [_, rapidpro_update, turn_archive] = responses.calls
+        self.assertEqual(
+            json.loads(rapidpro_update.request.body),
+            {
+                "fields": {
+                    "helpdesk_timeout": None,
+                    "wait_for_helpdesk": None,
+                    "helpdesk_message_id": None,
+                }
+            },
+        )
+        self.assertEqual(
+            json.loads(turn_archive.request.body),
+            {
+                "before": "ABGGJ4NjeFMfAgo-sCqKaSQU4UzP",
+                "reason": f"Auto archived after 11 days",
+            },
+        )
+
+    @responses.activate
+    def test_conversation_not_expired(self):
+        contact_id = "9e12d04c-af25-40b6-aa4f-57c72e8e3f91"
+
+        self.add_rapidpro_contact_list_response(
+            contact_id,
+            {
+                "helpdesk_timeout": "2020-01-09",
+                "wait_for_helpdesk": "TRUE",
+                "helpdesk_message_id": "ABGGJ4NjeFMfAgo-sCqKaSQU4UzP",
+            },
+        )
+
+        tasks.handle_expired_helpdesk_contacts()
+
+        self.assertEqual(len(responses.calls), 1)
+
+    @responses.activate
+    def test_conversation_fields_not_populated(self):
+        contact_id = "9e12d04c-af25-40b6-aa4f-57c72e8e3f91"
+
+        self.add_rapidpro_contact_list_response(
+            contact_id, {"wait_for_helpdesk": "TRUE", "helpdesk_message_id": None}
+        )
+
+        tasks.handle_expired_helpdesk_contacts()
+
+        self.assertEqual(len(responses.calls), 1)
+
+    @responses.activate
+    def test_conversation_expired_no_urn(self):
+        contact_id = "9e12d04c-af25-40b6-aa4f-57c72e8e3f91"
+
+        self.add_rapidpro_contact_list_response(
+            contact_id,
+            {
+                "helpdesk_timeout": "2020-01-06",
+                "wait_for_helpdesk": "TRUE",
+                "helpdesk_message_id": "ABGGJ4NjeFMfAgo-sCqKaSQU4UzP",
+            },
+            [],
+        )
+
+        self.add_rapidpro_contact_update_response(
+            contact_id,
+            {
+                "helpdesk_timeout": None,
+                "wait_for_helpdesk": None,
+                "helpdesk_message_id": None,
+            },
+            [],
+        )
+
+        tasks.handle_expired_helpdesk_contacts()
+
+        [_, rapidpro_update] = responses.calls
+        self.assertEqual(
+            json.loads(rapidpro_update.request.body),
+            {
+                "fields": {
+                    "helpdesk_timeout": None,
+                    "wait_for_helpdesk": None,
+                    "helpdesk_message_id": None,
```

Comment:
Just want to check, does doing a `POST` with `"field_name": null`, clear the field? I haven't tested that before.

Reply:
Yes it does clear it